File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* temp = head;
        if (head == nullptr) {
            return head;
        }
        ListNode* prev = NULL;
        ListNode* val;
        while(temp != NULL){
            val = temp->next;
            temp->next = prev;
            prev = temp;
            temp = val;
        }
        return prev;
    }
};

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} list1
     * @param {ListNode} list2
     * @return {ListNode}
     */
    mergeTwoLists(list1, list2) {
        var arr = []
        var temp1 = list1
        var temp2 = list2
        while(temp1 != null){
            arr.push(temp1.val);
            temp1 = temp1.next;
        }
        while(temp2 != null){
            arr.push(temp2.val)
            temp2 = temp2.next;
        }
        arr.sort((a, b) => a - b)
        const ans = new ListNode(0)
        let curr = ans
        for(let i = 0; i < arr.length; i++){
            curr.next = new ListNode(arr[i]);
            curr = curr.next
        }
        return ans.next
    }
}
